dramatiq/middleware/threading.py

```diff
@@ -0,0 +1,67 @@
+# This file is a part of Dramatiq.
+#
+# Copyright (C) 2017,2018 CLEARTYPE SRL <bogdan@cleartype.io>
+#
+# Dramatiq is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or (at
+# your option) any later version.
+#
+# Dramatiq is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+# License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import ctypes
+import platform
+import warnings
+
+from ..logging import get_logger
+from .middleware import Middleware
+
+
+class ThreadExceptingMiddleware(Middleware):
```

Comment:
I think it might be better not to rely on inheritance here. Just extract `raise_thread_exception` as a normal function in this module that the middleware can use. We can leave out the platform warning or we can have each of the middleware implement it individually, I don't see much value in DRYing that up, especially if it means having to follow the inheritance tree up in order to figure out where it's firing from.